core/commands/escrow/escrow.go

```diff
@@ -0,0 +1,149 @@
+	// TODO: Make escrow GuardAddress configurable
+	//authAddress := configuration.Services.GuardPubKeys[0]
+	authAddress := "guardAddress"
+	return &escrowpb.EscrowContract{
+		ContractId:       []byte(id),
+		BuyerAddress:     payerAddr,
+		SellerAddress:    recverAddr,
+		AuthAddress:      []byte(authAddress),
+		Amount:           price,
+		CollateralAmount: 0,
+		WithholdAmount:   0,
+		TokenType:        escrowpb.TokenType_BTT,
+		PayoutSchedule:   0,
+		NumPayouts:       1,
+	}
+}
+
+func NewSignedContract(contract *escrowpb.EscrowContract) *escrowpb.SignedEscrowContract {
+	return &escrowpb.SignedEscrowContract{
+		Contract: contract,
+	}
+}
+
+func NewContractRequest(configuration *config.Config, chunkInfo map[string]*storage.Chunk) (*escrowpb.EscrowContractRequest, error) {
+	var signedContracts []*escrowpb.SignedEscrowContract
+	var totalPrice int64
+	for _, chunk := range chunkInfo {
+		sc, err := UnmarshalEscrowContract(chunk.SignedContract)
+		if err != nil {
+			return nil, err
+		}
+		signedContracts = append(signedContracts, sc)
+		totalPrice += chunk.Price
+	}
+	// prepare channel commit
+	pid := configuration.Identity.PeerID
+	buyerPrivKey, err := configuration.Identity.DecodePrivateKey("")
+	if err != nil {
+		return nil, err
+	}
+	// TODO: Make escrow Address configurable
+	//escrowAddress := configuration.Services.EscrowPubKeys[0]
+	escrowAddress := "escrowAddress"
+	chanCommit := &ledger.ChannelCommit{
+		Payer:     &ledger.PublicKey{Key: []byte(pid)},
+		Recipient: &ledger.PublicKey{Key: []byte(escrowAddress)},
+		Amount:    totalPrice, // total amount in the contract request
+		PayerId:   time.Now().UnixNano(),
+	}
+	buyerChanSig, err := crypto.Sign(buyerPrivKey, chanCommit)
+	if err != nil {
+		return nil, err
+	}
+
+	signedChanCommit := &ledger.SignedChannelCommit{
+		Channel:   chanCommit,
+		Signature: buyerChanSig,
+	}
+	return &escrowpb.EscrowContractRequest{
+		Contract:     signedContracts,
+		BuyerChannel: signedChanCommit,
+	}, nil
+}
+
+func SubmitContractToEscrow(configuration *config.Config, request *escrowpb.EscrowContractRequest) error {
+	var conn *grpc.ClientConn
+	conn, err := grpc.Dial("52.15.101.94:50051", grpc.WithInsecure())
+	if err != nil {
+		return err
+	}
+	defer conn.Close()
+	client := escrowpb.NewEscrowServiceClient(conn)
+	response, err := client.SubmitContracts(context.Background(), request)
+	if err != nil {
+		return err
+	}
+	if response == nil {
+		return fmt.Errorf("escrow reponse is nil")
+	}
+	escrowPubkey, err := convertPubKey(configuration.Services.EscrowPubKeys[0])
+	if err != nil {
+		return err
+	}
+	ok, err := crypto.Verify(escrowPubkey, response.Result, response.EscrowSignature)
+	if err != nil || !ok {
+		return fmt.Errorf("verify escrow failed %v", err)
+	}
+	return nil
+}
+
+func convertPubKey(pubStr string) (ic.PubKey, error) {
+	raw, err := base64.StdEncoding.DecodeString(pubStr)
+	if err != nil {
+		return nil, err
+	}
+	return ic.UnmarshalSecp256k1PublicKey(raw)
```

Comment:
here after decode you can use the go-btfs-common ToPubKey I added yesterday.

Reply:
Do you think if common has toPubKey method taking input as string like toPrivKey do would be better? It will save unnecessary decode key part in the main logic here. I leave a comment suggesting to do so. but it didn't so I can't use it